### PR TITLE
fix: genesis bug in pool incentives linking NoLock gauges and PoolIDs

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Caught this bug while testing incentive distributions from the mainnet state exported edge net.

Observed the following in the logs:
```
4:32PM INF distributeInternal CL for pool id 1133 finished
4:32PM INF distributeInternal, CreateIncentiveRecord NoLock gauge gaugeId=34523 height=11732289 module=incentives poolId=1134 remainCoinPerEpoch={"amount":"41314029279","denom":"uosmo"}
4:32PM INF distributeInternal CL for pool id 1134 finished
4:32PM INF distributeInternal, CreateIncentiveRecord NoLock gauge gaugeId=34524 height=11732289 module=incentives poolId=1135 remainCoinPerEpoch={"amount":"108157658047","denom":"uosmo"}
4:32PM INF distributeInternal CL for pool id 1135 finished
4:32PM INF distributeInternal, CreateIncentiveRecord NoLock gauge gaugeId=34525 height=11732289 module=incentives poolId=1136 remainCoinPerEpoch={"amount":"14323634734","denom":"uosmo"}
4:32PM INF distributeInternal CL for pool id 1136 finished
4:32PM ERR no pool associated with gauge id (34562) and duration (0)
4:32PM ERR error in epoch hook no pool associated with gauge id (34562) and duration (0)
```
- Noted that the gauge with this ID is an external NoLock gauge on mainnet

After investigations, realized that pool incentive store indexes are not exported correctly. Currently, we only export enough pool ID and gauge links for the internal gauges.

However, all NoLock gauges have additional store indexes to connect to a pool. These are particularly necessary for the external NoLockGauges

### Solution

The solution was to:
- continue exporting internal gauge IDs and their respective pool IDs in `AnyPoolToInternalGauges` genesis field
- Introduce a new genesis entry called `ConcentratedPoolToNoLockGauges` that focuses on connecting CL pool IDs and their respective gauges where gauges can be both external and internal

## Testing and Verifying

- Added `TestImportExportGenesis_ExternalNoLock` that validates NoLock pool incentives store export and reimport in more detail

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

**Comment by p0mvn** [comment]
@devbot-wizard help

File: x/pool-incentives/keeper/genesis_test.go [hunk 1]
```diff
@@ -139,5 +134,57 @@ func (s *KeeperTestSuite) TestExportGenesis() {
...
+	// We expect internal gauge to be created first
+	internalGaugeID := externalGaugeID - 1
+
+	// Export genesis
+	export := s.App.PoolIncentivesKeeper.ExportGenesis(s.Ctx)
+
+	// Validate that only one link for internal gauges is created
+	s.Require().Equal(1, len(export.AnyPoolToInternalGauges.PoolToGauge))
+
+	// Validate that 2 links, one for external and one for internal gauge, are created
```
Comment (czarcas7ic) [inline x/pool-incentives/keeper/genesis_test.go:166]: Modified the comment to this as the last comment threw me off

File: proto/osmosis/pool-incentives/v1beta1/incentives.proto [hunk 2]
```diff
@@ -55,6 +55,10 @@ message PoolToGauge {
   ];
 }
 
-message PoolToGauges {
+message AnyPoolToInternalGauges {
   repeated PoolToGauge pool_to_gauge = 2 [ (gogoproto.nullable) = false ];
+}
+
+message ConcentratedPoolToNoLockGauges {
+  repeated PoolToGauge pool_to_gauge = 1 [ (gogoproto.nullable) = false ];
```
Comment (czarcas7ic) [inline proto/osmosis/pool-incentives/v1beta1/incentives.proto:63]: Not quite sure why these need their own message types when the underlying is the same, and the name can be changed in the actual message, but will not block on this and will not change myself in the event this was intentional and its more of a nit.,

**Review by czarcas7ic — APPROVED** [review]
Good find / change

**Pull request merged**